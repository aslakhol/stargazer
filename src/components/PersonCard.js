import React from 'react';
import {
  Card,
  CardImg,
  CardBody,
  CardTitle,
} from 'reactstrap';

const PersonCard = (props) => {
  const { person, imgUrl } = props;
  return (
    <div className="mb-3">
      <Card>
        <CardImg top width="100%" src={imgUrl} />
        <CardBody>
          <CardTitle>{person.name}</CardTitle>
        </CardBody>
      </Card>
    </div>
  );
};

export default PersonCard;
