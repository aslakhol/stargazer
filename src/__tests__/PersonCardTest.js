import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
// import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import PersonCard from '../components/PersonCard';

const mockPerson = {
  name: 'C-3PO',
  url: 'https://swapi.co/api/people/2/',
};

describe('<PersonCard />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const component = shallow(<PersonCard person={mockPerson} />);

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
