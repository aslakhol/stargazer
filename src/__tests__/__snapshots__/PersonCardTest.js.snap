// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonCard /> render() renders the component 1`] = `
<div>
  <Card
    tag="div"
  >
    <CardImg
      tag="img"
      top={true}
      width="100%"
    />
    <CardBody
      tag="div"
    >
      <CardTitle
        tag="h5"
      >
        C-3PO
      </CardTitle>
    </CardBody>
  </Card>
</div>
`;
